atomicapp/nulecule/lib.py

```diff
@@ -64,46 +63,21 @@ def load_config(self, config, ask=False, skip_asking=False):
             None
         """
         for param in self.params:
-            value = config.get(self.namespace, {}).get(param[NAME_KEY]) or \
-                config.get(GLOBAL_CONF, {}).get(param[NAME_KEY])
+            value = config.get(param[NAME_KEY], scope=self.namespace)
             if value is None and (ask or (
                     not skip_asking and param.get(DEFAULTNAME_KEY) is None)):
                 cockpit_logger.info("%s is missing in answers.conf." % param[NAME_KEY])
                 value = Utils.askFor(param[NAME_KEY], param)
             elif value is None:
                 value = param.get(DEFAULTNAME_KEY)
-            if config.get(self.namespace) is None:
-                config[self.namespace] = {}
-            config[self.namespace][param[NAME_KEY]] = value
+            config.set(param[NAME_KEY], value, scope=self.namespace, source='runtime')
```

Comment:
It's an assignment by reference, so we can move ``self.config = config`` to the top of the function.